web/tests/message_store.test.js

```diff
@@ -132,7 +132,7 @@ test("process_new_message", () => {
 
     assert.equal(message.reply_to, "bob@example.com,cindy@example.com");
     assert.equal(message.to_user_ids, "103,104");
-    assert.equal(message.display_reply_to, "Bob, Cindy");
+    assert.equal(message.display_reply_to, "Bob and Cindy");
```

Comment:
The `web/src/message_notifications.js` code path appears to use this field for non-display purposes to construct unique keys; I think that might now be incorrect

The `web/src/recent_view_ui.js` code also seems to sort by `message.display_reply_to`; does that seem correct to you? Might require some auditing.

Reply:
In ```web/src/message_notifications.js```, we can convert back the internationalized list to an array and then to a list using .join(", "), and then set the key.

Something like this :

```
const reconstructed_list = message.display_reply_to
                          .replace('and', '') 
                          .split(',')
                          .map(item => item.trim())
                          .join(", ");

key = reconstructedList;
```

I'm not sure about ```web/src/recent_view_ui.js```, as it uses ```message.display_reply_to``` for sorting and ```message.display_reply_to``` are already sorted name, with an extra ```and``` in it. Will it effect the sorting of multiple strings? 
What is your view about it?